store/src/java-test/com/zimbra/cs/service/mail/SendMsgTest.java

```diff
@@ -255,4 +263,60 @@ public void sendUpload() throws Exception {
         MimeMessage mm = msg.getMimeMessage();
         Assert.assertEquals("correct top-level MIME type", "multipart/alternative", new ZContentType(mm.getContentType()).getBaseType());
     }
+
+    @Test
+    public void testSendMailEventLoggerTest() throws Exception {
+
+        EventLogHandler.Factory mockFactory = Mockito.mock(EventLogHandler.Factory.class);
+        EventLogHandler mockHandler = Mockito.mock(EventLogHandler.class);
+        Mockito.doReturn(mockHandler).when(mockFactory).createHandler(Mockito.anyString());
+
+        EventLogger.registerHandlerFactory("testhandler", mockFactory);
+
+        EventLogger.ConfigProvider mockConfigProvider = Mockito.mock(EventLogger.ConfigProvider.class);
+
+        Multimap<String, String> mockConfigMap = ArrayListMultimap.create();
+        mockConfigMap.put("testhandler", "");
+        Mockito.doReturn(mockConfigMap.asMap()).when(mockConfigProvider).getHandlerConfig();
+
+        Mockito.doReturn(1).when(mockConfigProvider).getNumThreads(); //ensures sequential event processing
+
+        EventLogger.getEventLogger(mockConfigProvider).startupEventNotifierExecutor();
+
+        Account acct = Provisioning.getInstance().getAccountByName("test@zimbra.com");
+        Mailbox mbox = MailboxManager.getInstance().getMailboxByAccount(acct);
+
+        // first, add draft message
```

Comment:
Why do you need to create a draft to test logging of SENT events?

Reply:
Honestly, we had a hard time figuring out how to construct a proper SendMsgRequest otherwise, so this just copies send-from-draft logic from another test.